lib/propolis/src/hw/nvme/admin.rs

```diff
@@ -282,4 +293,56 @@ impl NvmeCtrl {
             }
         }
     }
+
+    /// Write result data from an admin command into host memory
+    ///
+    /// Returns `Some(())` if successful, else None
+    fn write_admin_result<T: Copy>(
+        prp: cmds::PrpIter,
+        data: &T,
+        mem: &MemCtx,
+    ) -> Option<()> {
```

Comment:
I'm not especially fond of using `Option<()>` here, but it allowed the logic to be more terse for now.  I'm working on some improvements to `PrpIter` and will revisit some of the error handling there.